subsys/net/l2/wifi/wifi_nm.c

```diff
@@ -117,7 +117,7 @@ int wifi_nm_register_mgd_type_iface(struct wifi_nm_instance *nm,
 	for (int i = 0; i < CONFIG_WIFI_NM_MAX_MANAGED_INTERFACES; i++) {
 		if (!nm->mgd_ifaces[i].iface) {
 			nm->mgd_ifaces[i].iface = iface;
-			nm->mgd_ifaces[i].type = (1 << type);
+			nm->mgd_ifaces[i].type = type;
```

Comment:
this too needs to be a separate commit and BTW, as said above this needs broader cleanup, I have a local branch but needs more work, but I am okay to push these in this PR as this is needed now.

Reply:
Should I make a separate commit, or is it okay to include it with the current one?

Reply:
A separate commit for both NM changes.

Reply:
Created the following https://github.com/zephyrproject-rtos/zephyr/pull/76887 to resolve comment